storage/innobase/row/row0ins.cc

```diff
@@ -3795,6 +3795,16 @@ row_ins_step(
 do_insert:
 	/* DO THE CHECKS OF THE CONSISTENCY CONSTRAINTS HERE */
 
+	DBUG_EXECUTE_IF("innodb_insert_fail",
+			{
+				if (0 == strcmp("test/t1", node->table->name.m_name)) {
+					err = DB_LOCK_WAIT_TIMEOUT;
+					DBUG_SET("-d,innodb_insert_fail");
+					goto error_handling;
+				} else if (0 == strcmp("test/t2", node->table->name.m_name)) {
+					err = DB_LOCK_WAIT_TIMEOUT;
+					goto error_handling;}});
+
 	err = row_ins(node, thr);
```

Comment:
using hard-coded table names here for the failure injection works, but is not elegant.
We have recently created similar failure injection for FK failure cases, where the failure injection happens for a given number of times, and execution succeeds after last failure. Please take a look at commit ea082421db04b34b13786f7a3ea57578a784d6f8 and DBUG_EXECUTE_IF implementation in storage/innobase/row/row0ins.cc. Same approach should work here.
The FK failure injection has also hard coded part to fail exactly 4 times, I assume this could be parametrized and enable the mtr script to decide how many failures are wanted (if we want to take this to the next level)

Reply:
Removed one of the two hard-coded table names and replaced the other hard-coded name with the option variable `wsrep_innodb_insert_fail_table`.

Reply:
Fixed this problem.